### PR TITLE
Add logging ability.

### PR DESCRIPTION
**Pull request by Bluebugs**

### CODE REVIEW

**Comment by Bluebugs** [comment]
This should help address #17 in a generic way.

**Comment by Bluebugs** [comment]
I did spend a little bit of time looking at a few log module for golang and there is no interface except at the function level which is why I settle for just going with a Printf style prototype that is supported by all the module I checked.

File: README.md [hunk 1]
```diff
@@ -82,6 +82,24 @@ If you desire a GUI element and visual integration with Fyne, you should check [
 
 To help you manage your key, sign binary and upload them to an online S3 bucket the `selfupdatectl` tool is provided. You can check its documentation [here](https://github.com/fynelabs/selfupdate/tree/main/cmd/selfupdatectl).
 
+## Loggin
```
Comment (andydotxyz) [inline README.md]: Maybe Logging?

File: README.md [hunk 2]
```diff
@@ -82,6 +82,24 @@ If you desire a GUI element and visual integration with Fyne, you should check [
...
+
+We provide three package wide variable: `LogError`, `LogInfo` and `LogDebug` that follow `log.Printf` API to provide an easy way to hook any logger in. To use it with go logger, you can just do
```
Comment (andydotxyz) [inline README.md]: "variables"

File: log.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package selfupdate
+
+// LogError will be called to log any reason that have prevented an executable update
+var LogError func(string, ...interface{}) = nil
```
Comment (andydotxyz) [inline log.go]: These ` = nil` should not be needeed, all variables default to the zero value, for a `func` that is `nil`

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Great addition - but some typos and unneeded variable initialisation

**Review by andydotxyz — APPROVED** [review]
Great thanks

**Pull request merged**